internal/server/files.go

```diff
@@ -0,0 +1,184 @@
+package server
+
+import (
+	"bufio"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/labstack/echo/v4"
+)
+
+type files struct{}
+
+type Resource struct {
+	RemoteIdentifier string `json:"remoteIdentifier"`
+}
+
+type ValetRequestParams struct {
+	Operation string `json:"operation"`
+	Resources []Resource
+}
+
+type ValetToken struct {
+	Authorization string `json:"authorization"`
+	FileId        string `json:"fileId"`
+}
+
+func (token *ValetToken) GetFilePath() string {
+	// TODO: Check format of fileId (Security)
+	// TODO: Allow custom path in config
+	// TODO: Subfolders for each user (Compatible format with official server)
+	return "/etc/standardfile/database/" + token.FileId
+}
+
+// Provides a valet token that is required to execute an operation
+func (h *files) ValetTokens(c echo.Context) error {
+	var params ValetRequestParams
+	if err := c.Bind(&params); err != nil {
+		return c.JSON(http.StatusBadRequest, err)
+	}
+
+	if len(params.Resources) != 1 {
+		return c.JSON(http.StatusBadRequest, "Multi file requests not supported")
```

Comment:
Use `sferror.New("Multi file requests not supported")` that implements the StandardNotes error JSON format.

Reply:
For the file API, StandardNotes expects:
```json
{
  "success": false,
  "message": "Failed to ..."
}
```

In `sferror` the `success` field seems to be currently missing.

Reply:
I fixed the error handling.